patches/tModLoader/Terraria/ModLoader/Default/UnloadedTilesSystem.cs

```diff
@@ -5,113 +5,49 @@
 namespace Terraria.ModLoader.Default
 {
 	[LegacyName("UnloadedTilesWorld")]
-	class UnloadedTilesSystem : ModSystem
-	{
-		internal List<UnloadedTileInfo> infos = new List<UnloadedTileInfo>();
-		internal List<UnloadedTileInfo> pendingInfos = new List<UnloadedTileInfo>();
+	internal class UnloadedTilesSystem : ModSystem {
+		/// <summary>
+		/// Tile-<see cref="UnloadedInfo"/>s that are not able to be restored in the current state of the world (and saved for the next world load)
+		/// </summary>
+		internal List<UnloadedInfo> tileInfos = new List<UnloadedInfo>();
+
+		/// <summary>
+		/// Wall-<see cref="UnloadedInfo"/>s that are not able to be restored in the current state of the world (and saved for the next world load)
+		/// </summary>
+		internal List<UnloadedInfo> wallInfos = new List<UnloadedInfo>();
 
 		public override void OnWorldLoad() {
-			infos.Clear();
-			pendingInfos.Clear();
+			tileInfos.Clear();
+			wallInfos.Clear();
 		}
 
 		public override TagCompound SaveWorldData() {
 			return new TagCompound {
-				["list"] = infos.Select(info => info?.Save() ?? new TagCompound()).ToList()
+				["tileInfos"] = tileInfos.Select(info => info?.Save() ?? new TagCompound()).ToList(),
+				["wallInfos"] = wallInfos.Select(info => info?.Save() ?? new TagCompound()).ToList()
 			};
 		}
 
 		public override void LoadWorldData(TagCompound tag) {
-			List<ushort> canRestore = new List<ushort>();
-			bool canRestoreFlag = false;
-
-			foreach (var infoTag in tag.GetList<TagCompound>("list")) {
-				if (!infoTag.ContainsKey("mod")) {
-					infos.Add(null);
-					canRestore.Add(0);
-					continue;
-				}
-
-				string modName = infoTag.GetString("mod");
-				string name = infoTag.GetString("name");
-				bool frameImportant = infoTag.ContainsKey("frameX");
-				var info = frameImportant ?
-					new UnloadedTileInfo(modName, name, infoTag.GetShort("frameX"), infoTag.GetShort("frameY")) :
-					new UnloadedTileInfo(modName, name);
-
-				infos.Add(info);
-
-				int type = ModContent.TryFind(modName, name, out ModTile tile) ? tile.Type : 0;
+			// Process tiles
+			tileInfos.AddRange(TileIO.tileInfos);
```

Comment:
I think `TileIO` should have a more active interface with this class. Rather than merging two separate lists in this hook

Reply:
Class is gone, so not applicable.